app/models/framework/definition/parser.rb

```diff
@@ -24,8 +24,9 @@ class Parser < Parslet::Parser
       rule(:management_charge)    { str('ManagementCharge') >> (column_based | flat_rate | sector_based).as(:management_charge) }
       rule(:flat_rate)            { (spaced(percentage).as(:value) >> flat_rate_column.maybe).as(:flat_rate) }
       rule(:flat_rate_column)     { spaced(str('of')) >> string.as(:column) }
-      rule(:column_based)         { spaced(str('varies_by')) >> (spaced(string).as(:column_name) >> spaced(dictionary).as(:value_to_percentage)).as(:column_based) }
+      rule(:column_based)         { (spaced(str('varies_by')) >> column_names.as(:column_names) >> multi_key_dictionary.as(:value_to_percentage)).as(:column_based) }
```

Comment:
I don't quite see why the opening bracket has moved...

Reply:
Probably happened when I was pulling out the `column_name` and `column_names` rules. I've moved it back and the tests still pass!